ParadiseSS13/Paradise
# Removes red slime speed potion, adds slime speed treat

## What Does This PR Do

- Removes the red slime speed potion
- Adds a slime speed treat
- Refactors some code to remove copy-paste

## Why It's Good For The Game

**Removes the red slime speed potion**

I believe it is a problematic item that causes a lot of issues when it comes to balancing armor. If a piece of gear has a slowdown, it (usually) has a very good reason for it; and if a hardsuit has no slowdown whatsoever, it is because the hardsuit itself is a rare one. 

Take the blood-red hardsuit for example. The user can remove its slowdown with a single click, but the trade-off is that it paints them a huge target in the eyes of security. The red potion removes such a constraint from _everything,_ it can be mass-produced, and heavily tips the balance in favour of anyone who has access to xenobiology versus those who do not (including the ERT).

In case of the latter, we could just make the ERT armor not slow them down, but then we make the ERT armor way stronger against other threats, like biohazards (where speed is not as important as against humanoid threats).

**Adds a slime speed treat**

To avoid just removing stuff from the game without contributing to it, introducing the **slime speed treat** (or monkey snacc), idea originally by @Edan52, I sprited it and coded it.

Adding water to red slime extracts, it results in a slime speed treat. You can tap a slime with it or if you are a slime, you can eat it.

It sets the slime's body temperature to 550 (allowing them to zoom around for about 20 seconds), sets them jittery for 15 seconds (it was a good treat!), and give the following visible message: `"As the grey baby slime (335) gobbles the slime speed treat, it starts buzzing with joyful energy!"`

The temperature gun can do the same, I am aware, but this is an alternative.

Feeding the slime:

https://user-images.githubusercontent.com/33333517/192590097-a070bc1b-3b6d-41c3-b7c5-73b19376dafa.mp4

Slimes can consume it themselves:

https://user-images.githubusercontent.com/33333517/192590534-707e9ec5-b5c6-4868-aef3-e5fba36288ed.mp4

**Refactors some code to remove copy-paste**

Removed some unnecessary `icon =` declarations, made a parent `attack()` proc to remove copy-paste from the other potions.

## Images of changes

**Monkey snacc**

![image](https://user-images.githubusercontent.com/33333517/192590656-4190b160-2e0c-48c9-9a6c-3cd877ff4208.png)

**Slime speed treat:**

Videos above

## Testing

Slime speed treat:

1. Spawn in as a scientist
2. Spawn a `/obj/item/slime_extract/red`
3. Fill syringe with water, add to extract
4. Tap slime with the treat or be a slime and click on it

Spawned the following and hit slimes with it, to ensure my refactor did not break anything:

- `/obj/item/slimepotion/slime/docility`
- `/obj/item/slimepotion/sentience`
- `/obj/item/slimepotion/slime/steroid`
- `/obj/item/slimepotion/slime/stabilizer`
- `/obj/item/slimepotion/slime/mutator`

## Changelog
:cl: Miraviel, CornMyCob
del: Removed the slime speed potion from red slimes.
add: Added slime speed treat in its place, add water to a red slime extract to spawn it. Humanoids can feed it to slimes or sentient slimes can click on it to eat it.
/:cl:
